cpp/dolfinx/refinement/ParallelRefinement.cpp

```diff
@@ -425,19 +425,25 @@ ParallelRefinement::partition(const std::vector<std::int64_t>& cell_topology,
   for (std::int32_t i = 0; i < num_local_cells; i++)
     global_cell_indices[i] = idx_global_offset + i;
 
-  Eigen::Map<const Eigen::Array<std::int64_t, Eigen::Dynamic, Eigen::Dynamic,
-                                Eigen::RowMajor>>
-      cells(cell_topology.data(), num_local_cells, num_vertices_per_cell);
-
   // Build mesh
   if (redistribute)
   {
-    return mesh::create(_mesh.mpi_comm(),
-                        graph::AdjacencyList<std::int64_t>(cells),
-                        _mesh.geometry().cmap(), _new_vertex_coordinates,
-                        mesh::GhostMode::none);
+    mesh::GhostMode ghost_mode = mesh::GhostMode::none;
+    if (num_ghost_cells > 0)
```

Comment:
I think this step requires communication - some processes might not have ghost cells.

Reply:
I don't believe this is a problem, but we could do a reduce here to be sure.
Maybe instead the user should supply a ghost mode for the refined mesh, rather than trying to infer it from the original.

Reply:
It might be unusual, but there are cases where it could happen, e.g. two disconnected bodies. These cases can arise is contact and other problems.